### PR TITLE
Substrate v3 upgrade

### PR DESCRIPTION
**Pull request by iorveth**
Full substrate v3 runtime upgrade of `olymia_audit2` branch (runtime, unit tests, benchmarking)
Closes https://github.com/Joystream/joystream/issues/2542

### CODE REVIEW

File: runtime-modules/council/src/lib.rs [hunk 1]
```diff
@@ -917,7 +916,7 @@ impl<T: Trait> Module<T> {
                 // clear candidate record and unlock their candidacy stake
                 Mutations::<T>::clear_candidate(&membership_id, &candidate);
 
-                (candidate, membership_id, now, 0.into()).into()
+                (candidate, membership_id, now, 0_u32.into()).into()
```
Comment (shamil-gadelshin) [inline runtime-modules/council/src/lib.rs]: Please, use the consistent style for constant types. In other pallets, the style is `0u32`.

File: runtime/src/weights/pallet_staking.rs [hunk 2]
```diff
@@ -58,6 +58,14 @@ impl pallet_staking::WeightInfo for WeightInfo {
             .saturating_add(DbWeight::get().reads(2 as Weight))
             .saturating_add(DbWeight::get().writes(2 as Weight))
     }
+    fn kick(k: u32) -> Weight {
+        (31_066_000 as Weight)
+            // Standard Error: 11_000
+            .saturating_add((17_754_000 as Weight).saturating_mul(k as Weight))
+            .saturating_add(RocksDbWeight::get().reads(2 as Weight))
```
Comment (shamil-gadelshin) [inline runtime/src/weights/pallet_staking.rs]: Build error here: `DbWeight` should be instead of `RocksDbWeight`

File: types/augment-codec/augment-api-tx.ts [hunk 3]
```diff
@@ -2872,7 +2872,7 @@ declare module '@polkadot/api/types/submittable' {
        * - `calls`: The calls to be dispatched from the same origin.
        *
        * If origin is root then call are dispatch without checking origin filter. (This includes
-       * bypassing `frame_system::Trait::BaseCallFilter`).
+       * bypassing `frame_system::Config::BaseCallFilter`).
```
Comment (shamil-gadelshin) [inline types/augment-codec/augment-api-tx.ts]: We shouldn't change the *.ts files.

File: types/augment/augment-api-tx.ts [hunk 4]
```diff
@@ -2872,7 +2872,7 @@ declare module '@polkadot/api/types/submittable' {
        * - `calls`: The calls to be dispatched from the same origin.
        *
        * If origin is root then call are dispatch without checking origin filter. (This includes
-       * bypassing `frame_system::Trait::BaseCallFilter`).
+       * bypassing `frame_system::Config::BaseCallFilter`).
```
Comment (shamil-gadelshin) [inline types/augment/augment-api-tx.ts]: We shouldn't change the *.ts files.

File: analyses/bench/Cargo.toml [hunk 5]
```diff
@@ -10,26 +10,26 @@ license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 
 [dependencies]
 log = "0.4.8"
-node-primitives = { package = 'node-primitives', git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca' }
-node-runtime = { package = 'node-runtime', git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca' }
-sc-cli = { package = 'sc-cli', git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca' }
-sc-client-api = { package = 'sc-client-api', git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca' }
-sp-runtime = { package = 'sp-runtime', git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca' }
-sp-state-machine = { package = 'sp-state-machine', git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca' }
+node-primitives = { package = 'node-primitives', git = 'https://github.com/paritytech/substrate.git', rev = '49a4103f4bfef55be20a5c6d26e18ff3003c3353' }
```
Comment (shamil-gadelshin) [inline analyses/bench/Cargo.toml]: We don't support this crate until we need another round of analysis.

File: node/src/command.rs [hunk 6]
```diff
@@ -1,48 +1,33 @@
...
-        "https://www.joystream.org/".into()
-    }
-
-    fn copyright_start_year() -> i32 {
-        2019
-    }
-
-    fn executable_name() -> String {
-        "joystream-node".into()
+        "Substrate Node".into()
```
Comment (shamil-gadelshin) [inline node/src/command.rs]: Please, restore the original properties of CLI.

File: node/src/service.rs [hunk 7]
```diff
@@ -470,316 +409,22 @@ pub fn new_light_base(
...
-        network,
-        transaction_pool,
-    ))
-}
-
-/// Builds a new service for a light client.
-pub fn new_light(config: Configuration) -> Result<TaskManager, ServiceError> {
-    new_light_base(config).map(|(task_manager, _, _, _, _)| task_manager)
-}
-
```
Comment (shamil-gadelshin) [inline node/src/service.rs:511]: Is there a reason for deleting these tests? They are very helpful for detecting node config problems.

File: pioneer/packages/joy-proposals/src/stories/data/ProposalPreviewList.mock.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@ import mockedProposal from './ProposalDetails.mock';
 //   {
```
Comment (shamil-gadelshin) [inline pioneer/packages/joy-proposals/src/stories/data/ProposalPreviewList.mock.ts]: We shouldn't change the *.ts files.

File: runtime/src/runtime_api.rs [hunk 9]
```diff
@@ -65,14 +66,8 @@ pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
...
 
-/// A Block signed with a Justification
-pub type SignedBlock = generic::SignedBlock<Block>;
-
-/// BlockId type as expected by this runtime.
-pub type BlockId = generic::BlockId<Block>;
-
 /// Unchecked extrinsic type as expected by this runtime.
-pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<AccountId, Call, Signature, SignedExtra>;
+pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<Address, Call, Signature, SignedExtra>;
```
Comment (shamil-gadelshin) [inline runtime/src/runtime_api.rs:70]: Will it work using just `AccountId`?

File: runtime/src/lib.rs [hunk 10]
```diff
@@ -414,31 +447,21 @@ impl pallet_im_online::Trait for Runtime {
...
-    pub OffencesWeightSoftLimit: Weight = Perbill::from_percent(60) * MaximumBlockWeight::get();
-}
-
-impl pallet_offences::Trait for Runtime {
+impl pallet_offences::Config for Runtime {
     type Event = Event;
     type IdentificationTuple = pallet_session::historical::IdentificationTuple<Self>;
     type OnOffenceHandler = Staking;
-    type WeightSoftLimit = OffencesWeightSoftLimit;
+    type WeightSoftLimit = ();
```
Comment (shamil-gadelshin) [inline runtime/src/lib.rs]: Should we remove this constant?

File: tests/integration-tests/.env [hunk 11]
```diff
@@ -42,7 +42,7 @@ RUNTIME_UPGRADE_PROPOSAL_STAKE = 100000
 # Validator count increment for Validator count test.
 VALIDATOR_COUNT_INCREMENT = 2
 # Constantinople runtime path
-RUNTIME_WASM_PATH = ../../target/release/wbuild/joystream-node-runtime/joystream_node_runtime.compact.wasm
```
Comment (shamil-gadelshin) [inline tests/integration-tests/.env]: Is it really the case? My local build contains the `joystream_node_runtime.compact.wasm`.

File: tests/network-tests/.env [hunk 12]
```diff
@@ -27,7 +27,7 @@ RUNTIME_UPGRADE_PROPOSAL_STAKE = 100000
 # Validator count increment for Validator count test.
 VALIDATOR_COUNT_INCREMENT = 2
 # Constantinople runtime path
-RUNTIME_WASM_PATH = ../../target/release/wbuild/joystream-node-runtime/joystream_node_runtime.compact.wasm
```
Comment (shamil-gadelshin) [inline tests/network-tests/.env]: Is it really the case? My local build contains the `joystream_node_runtime.compact.wasm`.

File: runtime/src/lib.rs [hunk 13]
```diff
@@ -111,63 +110,108 @@ pub fn native_version() -> NativeVersion {
...
+            weights.max_total = Some(MAXIMUM_BLOCK_WEIGHT);
+            // Operational transactions have some extra reserved space, so that they
+            // are included even if block reached `MAXIMUM_BLOCK_WEIGHT`.
+            weights.reserved = Some(
+                MAXIMUM_BLOCK_WEIGHT - NORMAL_DISPATCH_RATIO * MAXIMUM_BLOCK_WEIGHT
+            );
+        })
+        .avg_block_initialization(AVERAGE_ON_INITIALIZE_RATIO)
+        .build_or_panic();
+    pub const SS58Prefix: u8 = 42; // Ss58AddressFormat::SubstrateAccount
```
Comment (shamil-gadelshin) [inline runtime/src/lib.rs:147]: Shouldn't we pick some non-default value here?

File: runtime/README.md [hunk 14]
```diff
@@ -9,7 +9,7 @@ Joystream node execute the code's logic to validate transactions and blocks on t
 
 When building joystream-node as described in [../node/README.md](../node/README.md), in addition to the joystream-node binary being built the WASM blob artifact is produced in:
 
-`./target/release/wbuild/joystream-node-runtime/joystream_node_runtime.compact.wasm`
+`./target/release/wbuild/joystream-node-runtime/node_runtime.compact.wasm`
```
Comment (shamil-gadelshin) [inline runtime/README.md]: Is it really the case? My local build contains the `joystream_node_runtime.compact.wasm`.

File: runtime-modules/working-group/Cargo.toml [hunk 15]
```diff
@@ -43,4 +43,5 @@ std = [
 	'common/std',
 	'balances/std',
 	'staking-handler/std',
-]
+	'membership/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/working-group/Cargo.toml:46]: Why do we need it here?

File: runtime-modules/utility/Cargo.toml [hunk 16]
```diff
@@ -43,4 +43,5 @@ std = [
     'council/std',
     'balances/std',
     'sp-arithmetic/std',
+    'working-group/std'
```
Comment (shamil-gadelshin) [inline runtime-modules/utility/Cargo.toml]: Why do we need it here?

File: runtime-modules/storage/Cargo.toml [hunk 17]
```diff
@@ -36,4 +36,5 @@ std = [
 	'pallet-timestamp/std',
 	'working-group/std',
 	'common/std',
+	'membership/std'
```
Comment (shamil-gadelshin) [inline runtime-modules/storage/Cargo.toml]: Why do we need it here?

File: runtime-modules/staking-handler/Cargo.toml [hunk 18]
```diff
@@ -5,23 +5,25 @@ authors = ['Joystream contributors']
 edition = '2018'
 
 [dependencies]
-sp-std = { package = 'sp-std', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-frame-support = { package = 'frame-support', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-frame-system = { package = 'frame-system', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-sp-arithmetic = { package = 'sp-arithmetic', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-pallet-balances = { package = 'pallet-balances', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
+serde = { version = "1.0.101", optional = true, features = ["derive"] }
```
Comment (shamil-gadelshin) [inline runtime-modules/staking-handler/Cargo.toml]: Why do we need it here? It compiles without it.

File: runtime-modules/service-discovery/Cargo.toml [hunk 19]
```diff
@@ -34,4 +34,5 @@ std = [
 	'frame-system/std',
 	'sp-runtime/std',
 	'working-group/std',
+	'membership/std'
```
Comment (shamil-gadelshin) [inline runtime-modules/service-discovery/Cargo.toml]: Why do we need it here?

File: runtime-modules/referendum/Cargo.toml [hunk 20]
```diff
@@ -42,5 +42,5 @@ std = [
     'common/std',
     'staking-handler/std',
     'balances/std',
+    'membership/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/referendum/Cargo.toml:45]: Why do we need it here?

File: runtime-modules/proposals/engine/Cargo.toml [hunk 21]
```diff
@@ -50,4 +50,7 @@ std = [
 	'balances/std',
     'common/std',
     'staking-handler/std',
+	'council/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/engine/Cargo.toml:53]: Why do we need it here?

File: runtime-modules/proposals/discussion/Cargo.toml [hunk 22]
```diff
@@ -45,5 +45,8 @@ std = [
 	'frame-system/std',
     'common/std',
     'balances/std',
-    'sp-runtime/std'
+    'sp-runtime/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/discussion/Cargo.toml:48]: Why do we need it here?

File: runtime-modules/proposals/codex/Cargo.toml [hunk 23]
```diff
@@ -56,4 +56,7 @@ std = [
     'proposals-engine/std',
     'proposals-discussion/std',
     'constitution/std',
+    'referendum/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/codex/Cargo.toml]: Why do we need it here?

File: runtime-modules/forum/Cargo.toml [hunk 24]
```diff
@@ -51,4 +51,6 @@ std = [
 	'pallet-timestamp/std',
 	'common/std',
     'balances/std',
+	'membership/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/forum/Cargo.toml:54]: Why do we need it here?

File: runtime-modules/council/Cargo.toml [hunk 25]
```diff
@@ -43,6 +43,6 @@ std = [
     'staking-handler/std',
     'common/std',
     'balances/std',
+    'membership/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/council/Cargo.toml:46]: Why do we need it here?

File: runtime-modules/bounty/Cargo.toml [hunk 26]
```diff
@@ -49,5 +50,8 @@ std = [
 	'balances/std',
 	'sp-runtime/std',
 	'common/std',
+	'membership/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/bounty/Cargo.toml:53]: Why do we need it here?

File: runtime-modules/blog/Cargo.toml [hunk 27]
```diff
@@ -5,36 +5,39 @@ authors = ['Joystream contributors']
...
 	'codec/std',
 	'sp-std/std',
 	'frame-support/std',
 	'sp-runtime/std',
 	'frame-system/std',
     'common/std',
     'sp-arithmetic/std',
-    'balances/std'
+    'balances/std',
+	'membership/std',
```
Comment (shamil-gadelshin) [inline runtime-modules/blog/Cargo.toml:42]: Why do we need it here?

File: runtime-modules/blog/Cargo.toml [hunk 28]
```diff
@@ -5,36 +5,39 @@ authors = ['Joystream contributors']
 edition = '2018'
 
 [dependencies]
-sp-std = { package = 'sp-std', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-sp-runtime = { package = 'sp-runtime', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-frame-support = { package = 'frame-support', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-frame-system = { package = 'frame-system', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
-codec = { package = 'parity-scale-codec', version = '1.0.0', default-features = false, features = ['derive'] }
-sp-arithmetic = { package = 'sp-arithmetic', default-features = false, git = 'https://github.com/paritytech/substrate.git', rev = 'a200cdb93c6af5763b9c7bf313fa708764ac88ca'}
+serde = { version = "1.0.101", optional = true, features = ["derive"] }
```
Comment (shamil-gadelshin) [inline runtime-modules/blog/Cargo.toml:8]: Why do we need it here? It compiles without it.

File: joystream-node.Dockerfile [hunk 29]
```diff
@@ -21,7 +21,7 @@ FROM debian:stretch
 LABEL description="Joystream node"
 WORKDIR /joystream
 COPY --from=builder /joystream/target/release/joystream-node /joystream/node
-COPY --from=builder /joystream/target/release/wbuild/joystream-node-runtime/joystream_node_runtime.compact.wasm /joystream/runtime.compact.wasm
```
Comment (shamil-gadelshin) [inline joystream-node.Dockerfile]: Is it really the case? My local build contains the joystream_node_runtime.compact.wasm.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Huge work is done! There are several issues left to finish:
- compilation errors (including the release build with the `runtime-benchmarks` feature enabled)
- *.ts files were affected
- we should increase all affected crates` major versions on such a major change
- remove newly added entries in the `std` feature of the `Cargo.toml`  files
- please, restore the `node service tests` and ensure their success 
- it fails at the start - please fix it.

File: runtime-modules/blog/Cargo.toml (hunk 27 above)
Comment (iorveth) [reply runtime-modules/blog/Cargo.toml:42]: `benchmarking` requires this feature to compile.

File: runtime-modules/bounty/Cargo.toml (hunk 26 above)
Comment (iorveth) [reply runtime-modules/bounty/Cargo.toml:53]: `benchmarking` requires this feature to compile.

File: runtime-modules/council/Cargo.toml (hunk 25 above)
Comment (iorveth) [reply runtime-modules/council/Cargo.toml:46]: `benchmarking` requires this feature to compile.

File: runtime-modules/proposals/discussion/Cargo.toml (hunk 22 above)
Comment (iorveth) [reply runtime-modules/proposals/discussion/Cargo.toml:48]: `benchmarking` requires this feature to compile.

File: runtime-modules/proposals/codex/Cargo.toml (hunk 23 above)
Comment (iorveth) [reply runtime-modules/proposals/codex/Cargo.toml]: `benchmarking` requires this feature to compile.

File: runtime-modules/forum/Cargo.toml (hunk 24 above)
Comment (iorveth) [reply runtime-modules/forum/Cargo.toml:54]: `benchmarking` requires this feature to compile.

File: runtime-modules/bounty/Cargo.toml (hunk 26 above)
Comment (shamil-gadelshin) [reply runtime-modules/bounty/Cargo.toml:53]: The it should be added to the `runtime-benchmarks` feature list. It compiles without it. Am I missing something?

Comment (iorveth) [reply runtime-modules/bounty/Cargo.toml:53]: I couldn't compile it when run benchmark on pallet crate level. It compiles when you run build on ~/Joystream crate level though.

File: runtime-modules/council/src/lib.rs (hunk 1 above)
Comment (iorveth) [reply runtime-modules/council/src/lib.rs]: Done https://github.com/Joystream/joystream/pull/2558/commits/200f38ce880a027858003a353d2d38a10bb8c171

File: runtime/src/weights/pallet_staking.rs (hunk 2 above)
Comment (iorveth) [reply runtime/src/weights/pallet_staking.rs]: Done https://github.com/Joystream/joystream/pull/2558/commits/d685b66c824febb578d3ad6c33d2347702e84d5d

File: types/augment-codec/augment-api-tx.ts (hunk 3 above)
Comment (iorveth) [reply types/augment-codec/augment-api-tx.ts]: Done https://github.com/Joystream/joystream/pull/2558/commits/895f1f77703cf88f205075e107f71af362bd83e2

File: types/augment/augment-api-tx.ts (hunk 4 above)
Comment (iorveth) [reply types/augment/augment-api-tx.ts]: Done https://github.com/Joystream/joystream/pull/2558/commits/895f1f77703cf88f205075e107f71af362bd83e2

File: node/src/command.rs (hunk 6 above)
Comment (iorveth) [reply node/src/command.rs]: Done here https://github.com/Joystream/joystream/pull/2558/commits/7c6e48764907cda10842ea2a45711b1dcaf1915d

File: runtime/src/runtime_api.rs (hunk 9 above)
Comment (iorveth) [reply runtime/src/runtime_api.rs:70]: it would't work, i get multiple runtime errors.

File: runtime/src/lib.rs (hunk 10 above)
Comment (iorveth) [reply runtime/src/lib.rs]: I didn't find it in node-template. I restored it in case we introduced it by ourselves and it's needed.
Restored https://github.com/Joystream/joystream/pull/2558/commits/831899e778a890a59273fe155425f83f83ec6c23

File: runtime/src/lib.rs (hunk 13 above)
Comment (iorveth) [reply runtime/src/lib.rs:147]: Should we? I can't imagine how this can affect us, until we become a parachain.

File: runtime-modules/blog/Cargo.toml (hunk 28 above)
Comment (iorveth) [reply runtime-modules/blog/Cargo.toml:8]: Good point, removed https://github.com/Joystream/joystream/pull/2558/commits/dc47373761f1be7583cd028f2ef7d07d8cb6ede4

File: pioneer/packages/joy-proposals/src/stories/data/ProposalPreviewList.mock.ts (hunk 8 above)
Comment (iorveth) [reply pioneer/packages/joy-proposals/src/stories/data/ProposalPreviewList.mock.ts:4]: It's hard to see any changes here.

Comment (iorveth) [reply pioneer/packages/joy-proposals/src/stories/data/ProposalPreviewList.mock.ts]: Restored https://github.com/Joystream/joystream/pull/2558/commits/4f72dd0471bdfe61af616d7a06dfcf5f35afdc76

File: joystream-node.Dockerfile (hunk 29 above)
Comment (iorveth) [reply joystream-node.Dockerfile]: Seems like i spoiled this file with auto-replacement. Done https://github.com/Joystream/joystream/pull/2558/commits/485a925536448101aa9a96e40992a62983ba68a5

File: runtime/README.md (hunk 14 above)
Comment (iorveth) [reply runtime/README.md]: Seems like i spoiled this file with auto-replacement. Done 485a925

File: tests/integration-tests/.env (hunk 11 above)
Comment (iorveth) [reply tests/integration-tests/.env]: restored https://github.com/Joystream/joystream/pull/2558/commits/4db6ccd69561c6e952d4c8a5de8327ecaa2e5b2b

File: tests/network-tests/.env (hunk 12 above)
Comment (iorveth) [reply tests/network-tests/.env]: restored https://github.com/Joystream/joystream/pull/2558/commits/099047ce99d99282e652886cdcde7f39bbe17c76

File: analyses/bench/Cargo.toml (hunk 5 above)
Comment (iorveth) [reply analyses/bench/Cargo.toml]: ok, great

File: runtime-modules/staking-handler/Cargo.toml (hunk 18 above)
Comment (iorveth) [reply runtime-modules/staking-handler/Cargo.toml]: Removed

File: runtime-modules/proposals/engine/Cargo.toml (hunk 21 above)
Comment (iorveth) [reply runtime-modules/proposals/engine/Cargo.toml:53]: `benchmarking` requires this feature to compile, also i sometimes can't make a regular build in some cases as well.

File: runtime-modules/referendum/Cargo.toml (hunk 20 above)
Comment (iorveth) [reply runtime-modules/referendum/Cargo.toml:45]: `benchmarking` requires this feature to compile, also i sometimes can't make a regular build in some cases as well.

File: runtime-modules/service-discovery/Cargo.toml (hunk 19 above)
Comment (iorveth) [reply runtime-modules/service-discovery/Cargo.toml]: `benchmarking` requires this feature to compile, also i sometimes can't make a regular build in some cases as well.

File: runtime-modules/storage/Cargo.toml (hunk 17 above)
Comment (iorveth) [reply runtime-modules/storage/Cargo.toml]: `benchmarking` requires this feature to compile, also i sometimes can't make a regular build in some cases as well.

File: runtime-modules/utility/Cargo.toml (hunk 16 above)
Comment (iorveth) [reply runtime-modules/utility/Cargo.toml]: `benchmarking` requires this feature to compile, also i sometimes can't make a regular build in some cases as well.

File: runtime-modules/working-group/Cargo.toml (hunk 15 above)
Comment (iorveth) [reply runtime-modules/working-group/Cargo.toml:46]: `benchmarking` requires this feature to compile, also i sometimes can't make a regular build in some cases as well.

File: runtime-modules/staking-handler/Cargo.toml (hunk 18 above)
Comment (iorveth) [reply runtime-modules/staking-handler/Cargo.toml]: It compiles, but not in testing environment. Added this to dev dependencies.

File: runtime-modules/blog/Cargo.toml (hunk 28 above)
Comment (iorveth) [reply runtime-modules/blog/Cargo.toml:8]: It compiles, but not in testing environment. Added this to dev dependencies.

File: node/src/service.rs (hunk 7 above)
Comment (iorveth) [reply node/src/service.rs:511]: Restored tests here https://github.com/Joystream/joystream/pull/2558/commits/dcb876f8839a10d426ceb27e37fb5cff9dcd788a

File: runtime-modules/proposals/codex/src/lib.rs [hunk 30]
```diff
@@ -103,19 +103,19 @@ pub trait WeightInfo {
...
-    + staking::Trait
-    + proposals_engine::Trait
+/// 'Proposals codex' substrate module Config
+pub trait Config:
+    frame_system::Config
+    + proposals_engine::Config
+    + proposals_discussion::Config
+    + common::membership::Config
+    + staking::Config
+    + proposals_engine::Config
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/codex/src/lib.rs:115]: Why do we have a duplicate here?

File: runtime-modules/proposals/discussion/Cargo.toml (hunk 22 above)
Comment (shamil-gadelshin) [reply runtime-modules/proposals/discussion/Cargo.toml:48]: Referendum and council are not required. However, it's not clear why some pallets are required and some are not.

File: runtime-modules/referendum/Cargo.toml (hunk 20 above)
Comment (shamil-gadelshin) [reply runtime-modules/referendum/Cargo.toml:45]: Ok

File: runtime-modules/service-discovery/Cargo.toml (hunk 19 above)
Comment (shamil-gadelshin) [reply runtime-modules/service-discovery/Cargo.toml]: Ok

File: runtime-modules/storage/Cargo.toml (hunk 17 above)
Comment (shamil-gadelshin) [reply runtime-modules/storage/Cargo.toml]: Ok

File: runtime/src/lib.rs (hunk 13 above)
Comment (shamil-gadelshin) [reply runtime/src/lib.rs:147]: We are going to send the code to the Audit team. And they could also spot the default value.

File: runtime/src/runtime_api.rs (hunk 9 above)
Comment (shamil-gadelshin) [reply runtime/src/runtime_api.rs:70]: Ok

File: runtime-modules/proposals/codex/Cargo.toml (hunk 23 above)
Comment (shamil-gadelshin) [reply runtime-modules/proposals/codex/Cargo.toml]: Ok

File: runtime-modules/forum/Cargo.toml (hunk 24 above)
Comment (shamil-gadelshin) [reply runtime-modules/forum/Cargo.toml:54]: Ok

File: runtime-modules/council/Cargo.toml (hunk 25 above)
Comment (shamil-gadelshin) [reply runtime-modules/council/Cargo.toml:46]: Ok

File: runtime-modules/bounty/Cargo.toml (hunk 26 above)
Comment (shamil-gadelshin) [reply runtime-modules/bounty/Cargo.toml:53]: Ok

File: node/src/command.rs [hunk 31]
```diff
@@ -53,8 +38,12 @@ impl SubstrateCli for Cli {
         env!("CARGO_PKG_AUTHORS").into()
     }
 
-    fn native_runtime_version(_: &Box<dyn ChainSpec>) -> &'static RuntimeVersion {
-        &node_runtime::VERSION
+    fn support_url() -> String {
```
Comment (shamil-gadelshin) [inline node/src/command.rs:41]: Fill the proper value here, please.

File: node/src/command.rs [hunk 32]
```diff
@@ -53,8 +38,12 @@ impl SubstrateCli for Cli {
...
+        "support.anonymous.an".into()
+    }
+
+    fn copyright_start_year() -> i32 {
```
Comment (shamil-gadelshin) [inline node/src/command.rs:45]: Fill the proper value here, please.

File: node/src/command.rs [hunk 33]
```diff
@@ -1,43 +1,28 @@
...
-    fn support_url() -> String {
-        "https://www.joystream.org/".into()
-    }
-
-    fn copyright_start_year() -> i32 {
-        2019
-    }
-
-    fn executable_name() -> String {
         "joystream-node".into()
```
Comment (shamil-gadelshin) [inline node/src/command.rs]: Why did you change the 'impl_name' value?

File: runtime-modules/proposals/engine/Cargo.toml (hunk 21 above)
Comment (shamil-gadelshin) [reply runtime-modules/proposals/engine/Cargo.toml:53]: Ok

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Please, make further adjustments:
- rollback all changes to the `analyses` folder
- at least one 'node-service' test is failing
- there are multiple clippy issues (some of them are `runtime/weights` based - it makes sense to fix them on the `weights/mod.rs` level). I use your recommended version of the compiler: `nightly-2021-05-11`

File: node/src/command.rs (hunk 33 above)
Comment (iorveth) [reply node/src/command.rs]: Changes restored https://github.com/Joystream/joystream/pull/2558/commits/9f957769088c3a28cefc76713ad37eddc63d2ee9

File: node/src/command.rs (hunk 32 above)
Comment (iorveth) [reply node/src/command.rs:45]: done

File: node/src/command.rs (hunk 31 above)
Comment (iorveth) [reply node/src/command.rs:41]: done

File: runtime-modules/proposals/codex/src/lib.rs (hunk 30 above)
Comment (iorveth) [reply runtime-modules/proposals/codex/src/lib.rs:115]: Duplicate removed https://github.com/Joystream/joystream/pull/2558/commits/ff16b9a952475e890b41e236367174e275459e9d

**Comment by iorveth** [comment]
> rollback all changes to the analyses folder`
>
Done, `Cargo.lock` removed due to the compilation issue.

> there are multiple clippy issues (some of them are runtime/weights based - it makes sense to fix them on the weights/mod.rs level). I use your recommended version of the compiler: nightly-2021-05-11
> 
All clippy issues were fixed

**Comment by iorveth** [comment]
```
at least one 'node-service' test is failing
```

Node-service tests were fixed, lots of improvements to the `/node` implementation.

**Review by shamil-gadelshin — APPROVED** [review]
Most issues are fixed:
- `node/service/test_sync` test is failing because of the time out. However, the Parity Subtrate 3.0 node_cli version of the same test is also failing. It's unreasonable to fix their tests for the audit.
- SS58Prefix still defaults to 42. But we didn't register our own prefix yet.

LGTM

**Comment by shamil-gadelshin** [comment]
I found that one test is failing when we include the `runtime-benchmarks` feature. Please, fix it.
  `benchmarking::tests::test_fill_opening_worker`

**Comment by iorveth** [comment]
> test_fill_opening_worker

done

**Review by shamil-gadelshin — APPROVED** [review]
The test was fixed. Merging...
